lib/matchers/isoDate.js

```diff
@@ -1,7 +1,7 @@
 var _ = require('lodash');
 var factory = require('../factory');
 
-var dateTimeRegex   = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d\d\d)?(Z|[+-]\d\d:\d\d)?$/;
+var dateTimeRegex   = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d+)?(Z|[+-]\d\d:\d\d)?$/;
```

Comment:
Except the part where you've put `(.\d{3,4})` for the milliseconds, which should remain `(.\d+)` as @cysp put it.